packages/kbn-search-connectors/types/native_connectors.ts

```diff
@@ -757,15 +757,53 @@ export const NATIVE_CONNECTOR_DEFINITIONS: Record<string, NativeConnector | unde
         validations: [],
         value: '',
       },
-      token: {
+      auth_method: {
         default_value: null,
         depends_on: [],
+        display: DisplayType.DROPDOWN,
+        label: i18n.translate('searchConnectors.nativeConnectors.github.authMethod.label', {
+          defaultMessage: 'Authentication method',
+        }),
+        options: [
+          {
+            label: i18n.translate(
+              'searchConnectors.nativeConnectors.github.options.personalAccessToken',
+              {
+                defaultMessage: 'Personal access token',
+              }
+            ),
+            value: 'personal_access_token',
```

Comment:
`'personal_access_token'` and `github_app` etc. are referenced a lot, I wonder if we could make them constants?

Reply:
Done: https://github.com/elastic/kibana/pull/179236/commits/2ee72484369bb1d9562f563fa3c8e5e459ec216b